docs/enhancements/infrastructure-monitoring-service-api/alarms.md

```diff
@@ -0,0 +1,770 @@
+3. Response with `AlarmEventRecordModifications` and appropriate code
+
+### `alarmSubscriptions` family
+
+#### Steps for `/O2ims_infrastructureMonitoring/{apiVersion}/alarmSubscriptions` with GET
+1. Query the storage `alarm_subscription_info` (optionally using `?filter` param values)
+2. Response with a list of `AlarmSubscriptionInfo` and appropriate code
+
+#### Steps for `/O2ims_infrastructureMonitoring/{apiVersion}/alarmSubscriptions` with POST
+1. Client calls with `AlarmSubscriptionInfo` payload
+2. Validate the filter (e.g check if the columns actually exist)
+3. Insert `alarm_subscription_info`, for now we limit to 5 (if already 5, return with an error).
+4. Response with `AlarmSubscriptionInfo` and appropriate code
+
+#### Steps for `/O2ims_infrastructureMonitoring/{apiVersion}/alarmSubscriptions/{alarmSubscriptionId}` with GET
+1. Client calls with an `alarmSubscriptionId`
+2. Query the storage `alarm_subscription_info` table using `alarmSubscriptionId`
+3. Response with retrieved instance of `AlarmSubscriptionInfo` and appropriate code
+
+#### Steps for `/O2ims_infrastructureMonitoring/{apiVersion}/alarmSubscriptions/{alarmSubscriptionId}` with DELETE
+1. Client calls with an `alarmSubscriptionId`
+2. Delete the row `alarm_subscription_info` using `alarmSubscriptionId`
+3. No special response (only appropriate code)
+
+### `probableCause` family
+
+#### Steps for `/O2ims_infrastructureMonitoring/{apiVersion}/probableCause` with GET
+1. Query the tables `probable_causes` and `alarm_definitions` to fetch probableCause ID, alarm name and alarm description
+2. Response with a list of `probableCause` and appropriate code
+
+#### Steps for `/O2ims_infrastructureMonitoring/{apiVersion}/probableCause/{probableCauseId}` with GET
+1. Query the tables `probable_causes` and `alarm_definitions` using `probableCauseId` to fetch probableCause ID (should be same as probableCauseId from input ), alarm name and alarm description
+2. Response with retrieved instance of `probableCause` and appropriate code
+
+
+### `internal` family
+
+NOTE: These APIs are not exposed to client and available only internally 
+
+#### Steps for `internal/v1/caas-alerts/alertmanager`
+This API is activated only after configuring ACM alertmanager that can call back alarm service.
+
+All the alerts coming through this endpoint are of ResourceType that's "Logical"(kind) and "Compute"(class) which is
+effectively a cluster. For additional types of ResourceType we would like support in the future, 
+we can simply follow the same pattern
+i.e figure out the Kind and Class to get the ID + understand the source of Definitions + potentially open up another 
+endpoint if alert aggregator is not Alertmanager. 
+
+A minimal configuration
+```yaml
+route:
+  receiver: webhook_receiver
+receivers:
+  - name: webhook_receiver
+    webhook_configs:
+      - url: "o-ran-inventory-api-alarms.kubernetes.svc/internal/v1/caasAlerts/alertmanager" # this will be derived from deployment
```

Comment:
wouldn't the address/hostname part here be something like:  alarm-server.oran-o2ims.svc.cluster.local using the typical convention of <service>.<namespace>.svc.cluster.local.

Reply:
yep! updated